### PR TITLE
SOLR-11905 New Admin UI input boxes for JSON query and facet DSL

### PR DESCRIPTION
**Pull request by janhoy**
https://issues.apache.org/jira/browse/SOLR-11905

I did not bother updating any screenshots in RefGuide, since the existing images did not cover these parts anyway.

### CODE REVIEW

File: solr/webapp/web/partials/query.html [hunk 1]
```diff
@@ -359,6 +362,10 @@
         </div>
       </div>
 
+      <label for="json_query" title="JSON Request API">
+        JSON Query (<a href="https://solr.apache.org/guide/json-request-api.html" target="_blank">DSL documentation</a>)
```
Comment (epugh) [inline solr/webapp/web/partials/query.html:366]: what about just having a icon for the link to the docs...   One is "Json Facet DSL", the other is "DSL documentation"

**Comment by epugh** [comment]
I saw that we have `question-white.png` used in the schema page, maybe we want to use it here to signfiy more inforamtion and link to the docs?

**Review by epugh — APPROVED** [review]
LGTM.   Nit about icon for the docs...    Maybe you want to create a seperate ticket for reviewing and chagning all doc links to being behind the same ? icon.

File: solr/webapp/web/partials/query.html (hunk 1 above)
Comment (janhoy) [reply solr/webapp/web/partials/query.html:366]: Good idea. Do you know how to do it with a CSS class instead img tag?

Comment (janhoy) [reply solr/webapp/web/partials/query.html:366]: Ok, I made it, looks quite good:
Skjermbilde 2022-01-29 kl  13 32 53
I'd wish we had an "arrow" or "chain" icon that would more clearly suggest that a new browser window will open.

Comment (janhoy) [reply solr/webapp/web/partials/query.html:366]: The icon "book-open-text.png" shows an open book and is perhaps an alternative, suggesting a documentation link?

book-open-text

Comment (epugh) [reply solr/webapp/web/partials/query.html]: I would introduce the book only as part of a pass through all the various screens.

**Comment by epugh** [comment]
Ship 🚢 it!

**Pull request merged**